andela/Titan-Devs
# #163518694 Notifications for users

#### What does this PR do?
- Create notifications functionality. 
#### Description of Task to be completed?
- The user can receive in-app notification
- User can receive email notificatoin
#### How should this be manually tested?
- Clone the repository
- Check out this branch `ft-user-notifications-163518694`
- Set up the database 
- Change the node environment to the testing environment. 
- Run tests `yarn run test`.
- Consume different API using client tool (ex: postman).
   - Create two users
   - Follow each other 
   - Create an article 
   - Like or comment on the article
    - Use the second user to like or comment on the same article, or like the comment created. 
- Consume `GET: /api/v1/users/:id/notifications` You should see your notifications. 
- Consume `GET: /api/v1/users/:id/notifications/:notificationId` to see one notificaiton. 
-  Consume `Delete : /api/v1/users/:id/notifications/:notificationId` to delete one notificaiton. 
    
#### Any background context you want to provide?
Due to the fact, a user might have hundreds, and thousands of notifications, some functionality like editing all notifications at the same time are not included. We discussed that if we could integrate push notifications services, that when the functionality can be implemented. For now, when a notification is read, it is marked are read. 

#### What are the relevant pivotal tracker stories?
#163518694
